### PR TITLE
Use different version if Revise is tracking proto

### LINKED ISSUE #13: Incompatible with using Revise
using ProtoStucts when Revise is loaded errors on method deletion.

Can't reproduce yet. More info needed.

### PR DESCRIPTION
**Pull request by Tortar**
Alternative to #35. Look at https://discourse.julialang.org/t/is-there-a-way-to-know-if-revise-is-enabled/110379/6 for the limitations. Thanks to @Eben60 for the most part of the implementation (need to add him as a coauthor at least). It works fine on my machine

Fixes #13 
Closes #35

### CODE REVIEW

**Comment by Tortar** [comment]
I think this one can be merged @BeastyBlacksmith without problems because it doesn't change the interface

File: src/ProtoStruct.jl [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
+    f = splitpath(@__FILE__)[end]
+    watched_files = getproperty(Base.loaded_modules[revise_pkgid], :watched_files)
+    d in keys(watched_files) || return false
+    trackedfiles = watched_files[d].trackedfiles
+    return f in keys(trackedfiles)
+end
+
 macro proto(expr)
+    if checkrev()
+        @eval __module__ $(_proto(expr))
```
Comment (BeastyBlacksmith) [inline src/ProtoStruct.jl:17]: Can you explain to me what this is doing? Is it evaluating the struct definition in the ProtoStructs module?

Comment (Tortar) [reply src/ProtoStruct.jl:17]: It will re-evaluate in the module the `proto` definition resides, so it should be okay

**Review by BeastyBlacksmith — APPROVED** [review]
I am not a big fan of wrapping structs in modules for this. In theory this should't be necessary with this approach.
On the other hand I don't use Revise, so I trust you here.

**Pull request merged**